Metamask wallet connection and sign-in flow

AliBhutta2073 opened the pull request.

AliBhutta2073 committed:
fixed issue of tast undefined

AliBhutta2073 committed:
Integrated Metamask wallet

AliBhutta2073 committed:
allowed metamask and other wallets users to create proposal

AliBhutta2073 committed:
Done

- Context created for metamask
- Added signin in add vote flow
- Adjusted metamask signin flow in existing flow
- Modified code as per requirements

AliBhutta2073 committed:
signin flow

AliBhutta2073 committed:
fixed code issue of not executing query instead returning error

moneta commented on backend/src/socket/ndauConnect.js:
```diff
@@ -26,7 +26,7 @@ module.exports = (_io) => {
     console.log(socket.id);
```
Please just rename the file name ndauConnect to, says, walletConnect or something similar

moneta commented on frontend/src/Components/buttons/connect_metamask_button.jsx:
```diff
@@ -0,0 +1,128 @@
+                wallet_address:metamaskWeb3.walletAddress,
+            });
+
+            socket.on(
```
We should get rid of these sockets

moneta commented on frontend/src/Components/VoteButton/VoteButton.tsx:
```diff
@@ -104,44 +107,69 @@ const VoteButton = ({ dynamicClassName, allowVote, selectedVoteOption }: VoteBut
-        payload,
-        signature,
+        payload: _payload,
+        signature
```
This should be the signature against the complete payload, not the sign-in signature. The signature should use  typed struct V4

AliBhutta2073 committed:
Done

- Changed 'personal_sign' sign-in method to eth_signTypedData_v4 for metamask
- Fixed issue of undefined value for 'voting_option_heading' while selecting voting option

AliBhutta2073 committed:
Changed auth flow of wallet from sockets emitter/listener to direct wallet storing when authenticated via metamask

AliBhutta2073 commented:
Closing, will create a new PR with changes.
